### PR TITLE
nRF52833 support WIP

### PR DESCRIPTION
**Pull request by IanBallinger**
Runs on device hardware (not yet tested on pca10100). Flashing with nrfjprog 10.12.1 manually installed into BSP. 

Current issue: Bluefruit.begin() breaks when bond_init() calls InternalFS functions. 

Implementation borrows from work done by @LokiMetaSmith in PR #659.

### CODE REVIEW

**Comment by jpconstantineau** [comment]
I am really interested in this.  I made sure that CircuitPython support was working fine for the 833 but have been hoping for Arduino support for the 833 to come.  Debugging on CP pointed me to update the bootloader SD.  The UF2 device family is also different with the 833  (see bootloader repo)

Couple of comments:
1 - are the bootloader files the updated ones?
2 - will this support the 7.x SD that's in the newly updated bootloader? (the 6.x SD isn't supported by the 833)

**Comment by IanBallinger** [comment]
Per the [compatibility matrix](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fug_gsg_ses%2FUG%2Fgsg%2Fsoftdevices.html), the nRF52833 requires S140 7.0.1 or later. The current bootloader included in this fork was built with S140 6.1.1 for compatibility with the rest of the core since I think some function signatures change between versions (but i could be wrong). A softdevice update would probably be the last major thing needed to get board support working.

File: variants/pca10100/variant.h [hunk 1]
```diff
@@ -0,0 +1,155 @@
+ /*
```
Comment (IanBallinger) [inline variants/pca10100/variant.h:1]: file has not been double checked, it just follows the same structure as the other dev kit variants.

File: variants/pca10100/variant.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*
```
Comment (IanBallinger) [inline variants/pca10100/variant.cpp:1]: file has not been double checked, it just follows the same structure as the other dev kit variants.

File: cores/nRF5/nordic/nrfx_config.h [hunk 3]
```diff
@@ -26,7 +26,7 @@
 #define NRFX_TIMER2_ENABLED 0
 #define NRFX_TIMER3_ENABLED 0
 
-#ifdef NRF52840_XXAA
+#if defined(NRF52840_XXAA)
```
Comment (IanBallinger) [inline cores/nRF5/nordic/nrfx_config.h:29]: the nRF52833 does not have a QSPI peripheral

**Pull request closed without merging**